app/controllers/api/vacations_controller.rb

```diff
@@ -2,19 +2,20 @@
 
 module Api
   class VacationsController < Api::BaseController
-    before_action :authenticate_admin_or_manager_or_leader!, only: %i[vacation_applications show decline approve undone generate_csv]
-    before_action :find_vacation, only: %i[approve decline undone]
+    before_action :authenticate_admin_or_manager_or_leader!, only: %i[vacation_applications show decline approve undone generate_csv self_decline generate_yearly_report]
+    before_action :find_vacation, only: %i[approve decline undone self_decline]
     respond_to :json
 
     def index
       @vacations = Vacation.where('user_id = :user_id AND (extract(year from start_date) = :year OR extract(year from start_date) = :year)',
                                   user_id: current_user.id, year: params[:year]).order(:start_date)
-      @available_vacation_days = current_user.available_vacation_days
-      @used_vacation_days = current_user.used_vacation_days(@vacations)
+      @available_vacation_days = current_user.available_vacation_days(@vacations)
+      @used_vacation_days = current_user.used_vacation_days(@vacations, true)
     end
 
     def create
       @vacation = Vacation.create(vacations_params)
+      VacationMailer.send_information_to_accountancy(@vacation).deliver_later if @vacation
```

Comment:
I think this should be send only if record is valid. Check what will happen if someone sends invalid record i.e. end date < start date